Tests/scripts/utils/log_util.py

```diff
@@ -108,10 +114,198 @@ def install_simple_logging():
+        - Uses a different colored format for each log level (see LEVEL_STYLES)
+        - Defines a FileHandler with the given log_file_name.
+        - Defines a SteamHandler for stdout.
+        - Defines a logger for real time logs with those handlers
+        Args:
+            log_file_name: The path to where the log file will be saved
+        """
+        if not hasattr(logging, 'success'):
+            _add_logging_level('SUCCESS', 25)
+        self.log_file_name = log_file_name
+        formatter = coloredlogs.ColoredFormatter(fmt=LOGGING_FORMAT,
+                                                 level_styles=LEVEL_STYLES)
+        self.console_handler = logging.StreamHandler(sys.stdout)
+        self.console_handler.setFormatter(formatter)
+        log_file_path = os.path.join(ARTIFACTS_PATH, 'logs', log_file_name) if os.path.exists(
+            os.path.join(ARTIFACTS_PATH, 'logs')) else os.path.join(ARTIFACTS_PATH, log_file_name)
+        self.file_handler = logging.FileHandler(log_file_path)
+        self.file_handler.setFormatter(formatter)
+        self.console_handler.setLevel(logging.INFO)
+        self.file_handler.setLevel(logging.DEBUG)
+        self.real_time_logger = logging.getLogger(f'real_time-{log_file_path}')
+        self.real_time_logger.addHandler(self.file_handler)
+        self.real_time_logger.addHandler(self.console_handler)
+        self.real_time_logger.setLevel(logging.DEBUG)
+        self.loggers = {}
+        self.listeners = {}
+        self.logs_lock = Lock()
+        self.thread_names = set()
+
+    def _add_logger(self):
+        """
+        Defines a new logger, queueHandler and QueueListener for a new thread
+        This method is only called if the thread name is not in the 'thread_names' set.
+        """
+        thread_name = currentThread().getName()
+        log_queue = queue.Queue(-1)
+        queue_handler = QueueHandler(log_queue)
+        queue_handler.setLevel(logging.DEBUG)
+
+        logger = logging.getLogger(thread_name)
+        logger.setLevel(logging.DEBUG)
+        logger.addHandler(queue_handler)
+        listener = QueueListener(log_queue, self.console_handler, self.file_handler, respect_handler_level=True)
+        self.loggers[thread_name] = logger
+        self.listeners[thread_name] = listener
+        self.thread_names.add(thread_name)
+
+    def debug(self, message: str, real_time: bool = False) -> None:
+        """
+        Executes a debug log.
+        If real_time is given - will use the real_time_logger, if not - will use the thread's logger.
+        If the thread logger is used - the log will not be written until 'execute_logs' is called.
+        Args:
+            message: The log message
+            real_time: Whether to log the message now or after the next 'execute_logs' method execution.
+        """
+        thread_name = currentThread().getName()
+        if thread_name not in self.thread_names:
+            self._add_logger()
```

Comment:
pass the `thread_name` to the function so you don't need to process it twice

Reply:
Process what twice?

The whole point here is that I don't need to add any arguments such as thrad index or thread name.

Currently the thread_index is passed through all methods just to be able to print.

I want this logging manager to be a const in the file so that there will be no need to add any parameters or executions in order to log other then this method.

BTW `thread_name` is in the RAM and it's not taking any a few ns to retrive it

Reply:
i was refering to `thread_name = currentThread().getName()`.
you have it both here and inside `_add_logger`.